proposals/ssss-distributed-actor-runtime.md

```diff
@@ -1226,103 +1330,43 @@ extension ClusterSystem.ClusterTargetInvocation { // TODO: Split it into 2 separ
 
-#### Resolving the Recipient 
+#### Recipient: Resolving the recipient actor instance
+
+Now that we have prepared our `InvocationDecoder` we are ready to make the next step, and resolve the recipient actor which the invocation shall be made on. 
 
-The first step to prepare the invocation is to resolve the target actor. We discussed how resolving actors works in [Resolving Distributed Actors](#resolving-distributed-actors), however in this section we can tie it into the real process of invoking the target function as well.
+We already discussed how resolving actors works in [Resolving Distributed Actors](#resolving-distributed-actors), however in this section we can tie it into the real process of invoking the target function as well.
 
 In the example we're following so far, the recipient resolution is simple because we have the recipient id available in the `Envelope.recipientID`, so we only need to resolve that using the system that is receiving the message:
 
 ```swift
-let actor: DistributedActor = try self.resolveAny(id: envelope.recipientID) // where self is a DistributedActorSystem
+guard let actor: any DistributedActor = try self.knownActors[envelope.recipientID] else {
+  throw ClusterSystemError.unknownRecipient(envelope.recipientID)
+}
 ```
 
-The implementation here is the same as implementing the `DistributedActorSystem.resolve` just that it doesn't have to cast to a specific actor type -- it is not really necessary to perform the invocation. 
-
-If we needed to, we could confirm though if the actor is of the right type, by using the `actorType` of the resolved `DistributedFunction`, like this:
-
-```swift
-let actor: DistribtuedActor = try self.resolve(id: envelope.recipientID, as: dfunc.actorType)
-// which internally does:
-//   let anyActor = self.resolveAny(id: id)
-//   if anyActor as? dfunc.actorType else {
-//     throw ResolvedUnexpectedActorType(...) 
-//   }
-```
+This logic is the same as the internal implementation of the `resolve(id:as:)` method only that we don't have a need to validate the specific type of the actor - this will be handled by the Swift runtime in `executeDistributedTarget`'s implementation the target of the call which we'll explain in the next section.
 
-But since the returned value will end up erased anyway, it does not really matter all that much to try to resolve a specific type here, thus the above-mentioned `resolveAny`.
+#### Recipient: The `executeDistributedTarget` method
 
-#### The `executeDistributedTarget` method
+Invoking a distributed method is a tricky task, and involves a lot of type demangling, opening existential types, forming specific generic invocations and tightly managing all of that in order to avoid un-necessary heap allocations to pass the decoded arguments to the target function etc. After iterating over with multiple designs, we decided to expose a single `DistributedActorSystem.executeDistributedTarget` entry point which efficiently performs all the above operations. 
```

Comment:
Nit: You still need a dot after `etc.` to start a new sentence.

Reply:
Extraneous `with` after `over`?

Reply:
thanks!